freqtrade/freqtradebot.py

```diff
@@ -1192,7 +1205,7 @@ def execute_trade_exit(
             # Emergency sells (default to market!)
             order_type = self.strategy.order_types.get("emergencysell", "market")
 
-        amount = self._safe_exit_amount(trade.pair, trade.amount)
+        amount = sub_trade_amt or self._safe_exit_amount(trade.pair, trade.amount)
```

Comment:
you can do this IF you make sure that sub_trade_amt is really lower than trade amount (which i currently don't see anywhere).

you're also mixing stake amount with base amount (the return value of `adjust_trade_position` is in stake/quote amount).

Reply:
Updated.